### PR TITLE
Developer guide update with Spotless details

### PR DESCRIPTION
**Pull request by MitchellGale**
### Description
Adds more information and links about spotless and google java formatter in the developer guide.
 
### Check List
- [x] New functionality includes testing.
  - [x] All tests pass, including unit test, integration test and doctest
- [x] New functionality has been documented.
  - [x] New functionality has javadoc added
  - [x] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: DEVELOPER_GUIDE.rst [hunk 1]
```diff
@@ -182,17 +182,23 @@ Note that other related project code has already merged into this single reposit
 - ``workbench``: query workbench UI.
 
 
-Code Convention
+Code Convention (Spotless)
```
Comment (acarbonetto) [inline DEVELOPER_GUIDE.rst:185]: I wouldn't change this

File: DEVELOPER_GUIDE.rst [hunk 2]
```diff
@@ -182,17 +182,23 @@ Note that other related project code has already merged into this single reposit
...
+   * - Max line length is 100 characters (does not apply to import statements)
+   * - Line spacing is 2 spaces
+   * - Javadocs should be properly formatted in accordance to `Javadoc guidelines <https://www.oracle.com/ca-en/technical-resources/articles/java/javadoc-tool.html>`
+   * - Javadoc format can be maintained by wrapping javadoc with `<pre></pre>` HTML tags
+   * - Strings can be formatted on multiple lines with a `+` with the correct indentation for the string.
+
+Spotless changes required can be run with:
+`./gradlew spotlessJavaCheck`
+Recommended changes can be applied automatically with:
+`./gradlew spotlessJavaApply`
```
Comment (acarbonetto) [inline DEVELOPER_GUIDE.rst:201]: I would refer to the OpenSearch-project spotless documentation at the end (as a reference).  Something like: 

```
For more information on Spotless in the OpenSearch-project, please refer to the opensearch-project [developers_guide.md](link). 
```

File: DEVELOPER_GUIDE.rst [hunk 3]
```diff
@@ -182,17 +182,23 @@ Note that other related project code has already merged into this single reposit
...
 
-We’re integrated Checkstyle plugin into Gradle build: https://github.com/opensearch-project/sql/blob/main/config/checkstyle/google_checks.xml. So any violation will fail the build. You need to identify the offending code from Gradle error message and fix them and rerun the Gradle build. Here are the highlight of some Checkstyle rules:
-
-* 2 spaces indentation.
-* No line starts with tab character in source file.
-* Line width <= 100 characters.
-* Wildcard imports: You can enforce single import by configuring your IDE. Instructions for Intellij IDEA: https://www.jetbrains.com/help/idea/creating-and-optimizing-imports.html#disable-wildcard-imports.
-* Operator needs to wrap at next line.
-
+Java files are formatted using `Spotless <https://github.com/diffplug/spotless>` conforming to `Google Java Format <https://github.com/google/google-java-format>`.
```
Comment (acarbonetto) [inline DEVELOPER_GUIDE.rst:188]: https://sublime-and-sphinx-guide.readthedocs.io/en/latest/references.html#links-to-external-web-pages
needs `_`

File: DEVELOPER_GUIDE.rst [hunk 4]
```diff
@@ -182,17 +182,23 @@ Note that other related project code has already merged into this single reposit
...
+   * - New line at end of file
+   * - No unused import statements
+   * - Fix import order to be alphabetical with static imports first (one block for static and one for non-static imports)
+   * - Max line length is 100 characters (does not apply to import statements)
+   * - Line spacing is 2 spaces
+   * - Javadocs should be properly formatted in accordance to `Javadoc guidelines <https://www.oracle.com/ca-en/technical-resources/articles/java/javadoc-tool.html>`
```
Comment (acarbonetto) [inline DEVELOPER_GUIDE.rst:194]: https://sublime-and-sphinx-guide.readthedocs.io/en/latest/references.html#links-to-external-web-pages
needs `_`

Comment (MitchellGale) [reply DEVELOPER_GUIDE.rst]: Thanks, addressed

File: DEVELOPER_GUIDE.rst (hunk 3 above)
Comment (MitchellGale) [reply DEVELOPER_GUIDE.rst]: Thanks, addressed

File: DEVELOPER_GUIDE.rst (hunk 1 above)
Comment (MitchellGale) [reply DEVELOPER_GUIDE.rst]: Thanks, addressed

File: DEVELOPER_GUIDE.rst (hunk 2 above)
Comment (MitchellGale) [reply DEVELOPER_GUIDE.rst:201]: Leaving out as it is inconsistent with the exact formatting of our spotless java format application

File: DEVELOPER_GUIDE.rst (hunk 3 above)
Comment (acarbonetto) [reply DEVELOPER_GUIDE.rst]: Let's create an opening paragraph like https://github.com/opensearch-project/OpenSearch/blob/main/DEVELOPER_GUIDE.md#java-language-formatting-guidelines

How about: 
```
Java files in the OpenSearch codebase are formatted with the Eclipse JDT formatter, using the [Spotless Gradle](https://github.com/diffplug/spotless/tree/master/plugin-gradle) plugin. This plugin is configured in the project  `./gradle.properties`. 

The formatting check can be run explicitly with:
./gradlew spotlessJavaCheck

The code can be formatted with: 
./gradlew spotlessApply

These tasks can also be run for specific modules, e.g.
./gradlew server:spotlessJavaCheck

For more information on the spotless for the OpenSearch project please see `https://github.com/opensearch-project/OpenSearch/blob/main/DEVELOPER_GUIDE.md#java-language-formatting-guidelines`_.
```

Comment (acarbonetto) [reply DEVELOPER_GUIDE.rst]: Note: the only difference between what core uses and us is the gradle properties... which the core documentation stays should be configured on a project-by-project basis anyways.

File: DEVELOPER_GUIDE.rst (hunk 2 above)
Comment (MitchellGale) [reply DEVELOPER_GUIDE.rst]: Added with note about differences

File: DEVELOPER_GUIDE.rst (hunk 3 above)
Comment (MitchellGale) [reply DEVELOPER_GUIDE.rst]: Updated

File: DEVELOPER_GUIDE.rst [hunk 5]
```diff
@@ -185,14 +185,28 @@ Note that other related project code has already merged into this single reposit
...
-* Wildcard imports: You can enforce single import by configuring your IDE. Instructions for Intellij IDEA: https://www.jetbrains.com/help/idea/creating-and-optimizing-imports.html#disable-wildcard-imports.
-* Operator needs to wrap at next line.
+The formatting check can be run explicitly with:
+./gradlew spotlessJavaCheck
 
+The code can be formatted with:
+./gradlew spotlessApply
+
+These tasks can also be run for specific modules, e.g.
+./gradlew server:spotlessJavaCheck
```
Comment (Yury-Fridlyand) [inline DEVELOPER_GUIDE.rst:197]: You can emphasize this with a code block
```suggestion
These tasks can also be run for specific modules, e.g.::

./gradlew server:spotlessJavaCheck
```

File: DEVELOPER_GUIDE.rst [hunk 6]
```diff
@@ -185,14 +185,28 @@ Note that other related project code has already merged into this single reposit
...
-* 2 spaces indentation.
-* No line starts with tab character in source file.
-* Line width <= 100 characters.
-* Wildcard imports: You can enforce single import by configuring your IDE. Instructions for Intellij IDEA: https://www.jetbrains.com/help/idea/creating-and-optimizing-imports.html#disable-wildcard-imports.
-* Operator needs to wrap at next line.
+The formatting check can be run explicitly with:
+./gradlew spotlessJavaCheck
 
+The code can be formatted with:
+./gradlew spotlessApply
```
Comment (Yury-Fridlyand) [inline DEVELOPER_GUIDE.rst:194]: You can emphasize this with a code block
```suggestion
The code can be formatted with::

./gradlew spotlessApply
```

File: DEVELOPER_GUIDE.rst [hunk 7]
```diff
@@ -185,14 +185,28 @@ Note that other related project code has already merged into this single reposit
...
-We’re integrated Checkstyle plugin into Gradle build: https://github.com/opensearch-project/sql/blob/main/config/checkstyle/google_checks.xml. So any violation will fail the build. You need to identify the offending code from Gradle error message and fix them and rerun the Gradle build. Here are the highlight of some Checkstyle rules:
+Java files in the OpenSearch codebase are formatted with the Eclipse JDT formatter, using the [Spotless Gradle](https://github.com/diffplug/spotless/tree/master/plugin-gradle) plugin. This plugin is configured in the project  `./gradle.properties`.
 
-* 2 spaces indentation.
-* No line starts with tab character in source file.
-* Line width <= 100 characters.
-* Wildcard imports: You can enforce single import by configuring your IDE. Instructions for Intellij IDEA: https://www.jetbrains.com/help/idea/creating-and-optimizing-imports.html#disable-wildcard-imports.
-* Operator needs to wrap at next line.
+The formatting check can be run explicitly with:
+./gradlew spotlessJavaCheck
```
Comment (Yury-Fridlyand) [inline DEVELOPER_GUIDE.rst:191]: You can emphasize this with a code block
```suggestion
The formatting check can be run explicitly with::

./gradlew spotlessJavaCheck
```

**Review by Yury-Fridlyand — APPROVED** [review]

**Review by acarbonetto — APPROVED** [review]

**Pull request merged**